.github/workflows/ci_pull_request_open.yml

```diff
@@ -0,0 +1,62 @@
+name: Pull Request Open
+on:
+  pull_request:
+jobs:
+  ci-build-test-pr:
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx1500m -XX:+HeapDumpOnOutOfMemoryError"
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v5
+        with:
+          maven-version: 3.9.6
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: temurin
+          cache: maven
+
+      - run: mvn --version
+
+      - run: >
+            mvn -s maven-settings.xml clean install -DskipTests
+
+      - name: Test with Maven
+        run: >
+            mvn -s maven-settings.xml verify -B -e -DrerunFailingTestsCount=2
```

Comment:
We need to add `-Pnative` to the args here so that the native tests are executed. This is the cause of the "Infinispan Hot Rod Client GraalVM" module failures.

In order for native compilation to work, we need to add the following step to configure GraalVM:

```
      - uses: graalvm/setup-graalvm@v1
        with:
          java-version: '21'
          distribution: 'graalvm'
          github-token: ${{ secrets.GITHUB_TOKEN }}
```